intake_geopandas/geopandas.py

```diff
@@ -69,3 +47,78 @@ def to_dask(self):
 
     def _close(self):
         self._dataframe = None
+
+class GeoPandasFileSource(GeoPandasSource):
+    def __init__(self, urlpath, bbox=None, geopandas_kwargs=None, metadata=None):
+        """
+        Parameters
+        ----------
+        urlpath : str or iterable, location of data
+            Either the absolute or relative path to the file or URL to be opened.
+            Some examples:
+            - ``{{ CATALOG_DIR }}data/states.shp``
+            - ``http://some.domain.com/data/states.geo.json``
+        bbox : tuple | GeoDataFrame or GeoSeries, default None
+            Filter features by given bounding box, GeoSeries, or GeoDataFrame.
+            CRS mis-matches are resolved if given a GeoSeries or GeoDataFrame.
+        geopandas_kwargs : dict
+            Any further arguments to pass to geopandas's read_file function.
+        """
+        self.urlpath = urlpath
+        self._bbox = bbox
+        self._geopandas_kwargs = geopandas_kwargs or {}
+        self._dataframe = None
+
+        super().__init__(metadata=metadata)
+
+    def _open_dataset(self):
+        """
+        Open dataset using geopandas and use pattern fields to set new columns.
+        """
+        self._dataframe = geopandas.read_file(
+            self.urlpath, bbox=self._bbox, **self._geopandas_kwargs)
+
+class GeoJSONSource(GeoPandasFileSource):
+    name="geojson"
+
+class ShapefileSource(GeoPandasFileSource):
+    name="shapefile"
+
+class GeoPandasSQLSource(GeoPandasSource):
```

Comment:
Missing `name` attribute.

Reply:
The intention here was that this not actually be instantiated, and that users either select a postgis or sptialite source. It may end up being a distinction without a difference, but I kind of liked being more explicit about the source type. What do you think?

Reply:
You may want to make sure that the correct classes and names end up in `intake.registry`. That could be as simple as not exposing classes you don't intend for end-users in the top level of the package.

Reply:
It seems to be discovered by the registry as intended. The `GeoPandasSQLSource` is not exported in the `__init__.py` above.